src/main/scala/encry/consensus/PowLinearController.scala

```diff
@@ -22,11 +26,19 @@ object PowLinearController {
       }
       val diff: Difficulty = interpolate(data)
       if (diff >= chainParams.InitialDifficulty) DifficultySerializer.encodeCompactBits(diff) else chainParams.InitialNBits
-    } else previousHeaders.head._2.nBits
-  }
+    }
+
+  /** Used to provide `getDifficulty()` with the sequence of headers of correct heights. */
+  def getHeightsForRetargetingAt(height: Height): Seq[Height] = {
```

Comment:
Is it possible to replace this to
`     def getHeightsForRetargetingAt(height: Height): Seq[Height] = {
    if ((height - 1) % chainParams.EpochLength == 0)
      (0 to chainParams.RetargetingEpochsQty).reverse.map(i => (height - 1) - i *     chainParams.EpochLength).filter(_ >= Constants.Chain.GenesisHeight)
    else Seq(height - 1)
  }.map(i => Height @@ i)` ?

Reply:
If remove copypaste, logic will be more clear 
`(height - 1) % chainParams.EpochLength == 0`
`(0 to chainParams.RetargetingEpochsQty).reverse.map(i => (height - 1) - i * chainParams.EpochLength)`

Reply:
without filter